src/Tasks/AssemblyDependency/Reference.cs

```diff
@@ -1107,6 +1117,9 @@ string executableExtension
             // This is an assembly file, so we'll need to find dependencies later.
             DependenciesFound = false;
 
+            string externallyResolved = sourceItem.GetMetadata("ExternallyResolved");
+            ExternallyResolved = string.Equals(externallyResolved, "true", StringComparison.OrdinalIgnoreCase);
+
```

Comment:
Nit: Can you use:
`ExternallyResolved = MetadataConversionUtilities.TryConvertItemMetadataToBool(sourceItem, "ExternallyResolved")`

Not used everywhere, but it already is elsewhere in the file so we should try to be consistent.

Reply:
That looks like it would be an order of magnitude more expensive, but once per reference is probably ok. I'll do it for consistency.